### PR TITLE
Add RcppEnsmallen

### PR DESCRIPTION
**Pull request by coatless**
cc @conda-forge/r

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Related to: mlpack/ensmallen#235

Created using: https://github.com/bgruening/conda_r_skeleton_helper

### CODE REVIEW

**Comment by coatless** [comment]
Not sure what's going on here. I think the CI itself is having a spurious error moment. The extra packages that were added are overkill according to how dependencies have been structured on `RcppArmadillo`, e.g.

- https://github.com/conda-forge/r-mess-feedstock/blob/59feffae6aa443277ffe38eab334cb84c6d00656/recipe/meta.yaml#L23-L38
- https://github.com/conda-forge/r-biglasso-feedstock/blob/5a1f991a570f4a139c52d5ded6933b8a23e4effa/recipe/meta.yaml#L22-L40

**Comment by coatless** [comment]
So, if there is a dependency on `r-rcpparmadillo` any subsequent recipes require listing `_openmp_mutex`, `llvm-openmp`, `libblas`, and `liblapack` under `host:`, c.f.

```yaml
  host:
    - _openmp_mutex                # [linux]
    - llvm-openmp                  # [osx]
    - r-base
    - r-rcpp
    - r-rcpparmadillo >=0.8.400.0.0
    - libblas
    - liblapack
```

File: recipes/r-rcppensmallen/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    - {{ compiler('cxx') }}            # [not win]
+    - {{ compiler('m2w64_cxx') }}      # [win]
+    - {{ posix }}filesystem        # [win]
+    - {{ posix }}make
+    - {{ posix }}sed               # [win]
+    - {{ posix }}coreutils         # [win]
+    - {{ posix }}zip               # [win]
+
+  host:
+    - _openmp_mutex                # [linux]
```
Comment (dbast) [inline recipes/r-rcppensmallen/meta.yaml:36]: the mutex should come automatically via run_exports ... can you try removing it here?

Comment (coatless) [reply recipes/r-rcppensmallen/meta.yaml:36]:
```suggestion
```

Comment (coatless) [reply recipes/r-rcppensmallen/meta.yaml]: @dbast from the CI, removing this line caused the linux build to fail. Would you like me to re-add it?

**Comment by dbast** [comment]
@coatless Thanks!

**Pull request merged**